File: Android/scenes/ktv_singbattle/src/main/java/io/agora/scene/ktv/singbattle/create/holder/RoomHolder.java
package io.agora.scene.ktv.singbattle.create.holder;

import android.view.View;

import androidx.annotation.NonNull;

import io.agora.scene.base.GlideApp;
import io.agora.scene.base.component.BaseRecyclerViewAdapter;
import io.agora.scene.ktv.singbattle.R;
import io.agora.scene.ktv.singbattle.databinding.ItemRoomListBinding;
import io.agora.scene.ktv.singbattle.service.RoomListModel;
import io.agora.scene.widget.utils.CenterCropRoundCornerTransform;

public class RoomHolder extends BaseRecyclerViewAdapter.BaseViewHolder<ItemRoomListBinding, RoomListModel> {

    public RoomHolder(@NonNull ItemRoomListBinding mBinding) {
        super(mBinding);
    }

    @Override
    public void binding(RoomListModel data, int selectedIndex) {
        if (data != null) {
            GlideApp.with(mBinding.ivRoomCover.getContext()).load(data.getCoverRes())
                    .transform(new CenterCropRoundCornerTransform(40)).into(mBinding.ivRoomCover);
            mBinding.tvRoomName.setText(data.getName());
            mBinding.tvPersonNum.setText(String.format("%d%s", data.getRoomPeopleNum(), itemView.getContext().getString(R.string.ktv_people)));
            if (data.isPrivate()){
                mBinding.ivLock.setVisibility(View.VISIBLE);
            } else{
                mBinding.ivLock.setVisibility(View.GONE);
            }
        }
    }
}

File: Android/scenes/grandchorus/src/main/java/io/agora/scene/grandchorus/ui/holder/RoomHolder.kt
package io.agora.scene.grandchorus.ui.holder

import android.view.View
import io.agora.scene.base.GlideApp
import io.agora.scene.base.component.BaseRecyclerViewAdapter.BaseViewHolder
import io.agora.scene.grandchorus.R
import io.agora.scene.grandchorus.databinding.GrandchorusItemRoomListBinding
import io.agora.scene.grandchorus.service.RoomListModel
import io.agora.scene.widget.utils.CenterCropRoundCornerTransform

class RoomHolder constructor(mBinding: GrandchorusItemRoomListBinding) :
    BaseViewHolder<GrandchorusItemRoomListBinding, RoomListModel?>(mBinding) {
    override fun binding(data: RoomListModel?, selectedIndex: Int) {
        if (data != null) {
            GlideApp.with(mBinding.ivRoomCover.context).load(data.getCoverRes())
                .transform(CenterCropRoundCornerTransform(40)).into(mBinding.ivRoomCover)
            mBinding.tvRoomName.text = data.name
            mBinding.tvPersonNum.text =
                String.format("%d%s", data.roomPeopleNum, itemView.context.getString(R.string.grandchorus_people))
            if (data.isPrivate) {
                mBinding.ivLock.visibility = View.VISIBLE
            } else {
                mBinding.ivLock.visibility = View.GONE
            }
        }
    }
}